packages/optimizer/lib/transformers/OptimizeHeroImages.js

```diff
@@ -317,12 +322,21 @@ class OptimizeHeroImage {
       return;
     }
     node.attribs['i-amphtml-ssr'] = '';
-    node.attribs['data-hero'] = '';
+
     // Create img node
     const imgNode = createElement('img', {
       class: 'i-amphtml-fill-content i-amphtml-replaced-content',
       decoding: 'async',
     });
+
+    // If the image was detected as hero image candidate (and thus lacks an explicit
+    // data-hero), mark it as a hero and add loading=lazy to guard against making
+    // the page performance even worse by eagerly loading an image outside the viewport.
+    if (!('data-hero' in node.attribs)) {
+      node.attribs['data-hero'] = '';
+      imgNode.attribs['loading'] = 'lazy';
```

Comment:
I really like this approach. Have you checked if it is valid AMP?

Reply:
Yes, it is. The `loading` attribute is allowed, and it can be either `lazy` or `eager`: https://github.com/ampproject/amphtml/blob/a8889ac1b2622c461422a2186366f7a433848f44/validator/validator-main.protoascii#L3834-L3838

Reply:
Seems to be, otherwise the end-to-end spec would fail.